### ISSUE TITLE
bridge-nf-call-iptables is disabled in docker-dev container but is enabled on the host

### ISSUE THREAD

**Issue by tswift242** [docker/docker#27200]
**Description**

When building and starting docker within the docker-dev container, the output of docker info shows that bridge-nf-call-iptables and bridge-nf-call-ip6tables are disabled. However, these are both enabled on the host. Inside the docker-dev container, not only do those 2 files not exist, but the whole /proc/sys/net/bridge directory does not exist. On the host, both files exist and have value 1. There are also a lot of other files missing from /proc/sys/net/**.

A side effect (I believe) is that I'm not able to checkpoint a container (when I build with DOCKER_EXPERIMENTAL=1), using the basic example from the doc: 

```
docker run --security-opt=seccomp:unconfined --name cr -d busybox /bin/sh -c 'i=0; while true; do echo $i; i=$(expr $i + 1); sleep 1; done'
docker checkpoint create cr checkpoint1
```

This is the dump log: 
[dump.txt](https://github.com/docker/docker/files/514590/dump.txt)

**Steps to reproduce the issue:**
1. Steps 2-6 under Task 2 here: https://docs.docker.com/opensource/project/set-up-dev-env/
2. docker info

**Describe the results you received:**

Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: 1.13.0-dev
Storage Driver: overlay
 Backing Filesystem: extfs
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 4.4.0-0.bpo.tmw1-amd64
Operating System: Debian GNU/Linux 8 (jessie) (containerized)
OSType: linux
Architecture: x86_64
CPUs: 12
Total Memory: 62.83 GiB
Name: e860e34d4f9e
ID: SKHI:5K54:BIRR:BZNF:JUEB:LEVT:YAQL:ABM5:CIY3:7LZA:PVQW:NW52
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 13
 Goroutines: 21
 System Time: 2016-10-06T18:46:19.748657427Z
 EventsListeners: 0
Registry: https://index.docker.io/v1/
WARNING: No memory limit support
WARNING: No swap limit support
WARNING: No kernel memory limit support
WARNING: No oom kill disable support
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

Note that the first 4 warnings also appear on the host.

**Describe the results you expected:**

bridge-nf-call-iptables and bridge-nf-call-ip6tables should be enabled like they are on the host.

**Output of `docker version`:**

```
Client:
 Version:      1.13.0-dev
 API version:  1.25
 Go version:   go1.7.1
 Git commit:   f19120c
 Built:        Thu Oct  6 18:37:43 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.13.0-dev
 API version:  1.25
 Go version:   go1.7.1
 Git commit:   f19120c
 Built:        Thu Oct  6 18:37:43 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 0
Server Version: 1.13.0-dev
Storage Driver: overlay
 Backing Filesystem: extfs
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 4.4.0-0.bpo.tmw1-amd64
Operating System: Debian GNU/Linux 8 (jessie) (containerized)
OSType: linux
Architecture: x86_64
CPUs: 12
Total Memory: 62.83 GiB
Name: e860e34d4f9e
ID: SKHI:5K54:BIRR:BZNF:JUEB:LEVT:YAQL:ABM5:CIY3:7LZA:PVQW:NW52
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 13
 Goroutines: 21
 System Time: 2016-10-06T18:46:19.748657427Z
 EventsListeners: 0
Registry: https://index.docker.io/v1/
WARNING: No memory limit support
WARNING: No swap limit support
WARNING: No kernel memory limit support
WARNING: No oom kill disable support
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Physical host

**Comment by tswift242** [comment]
Quick update: I see the same error from the criu dump log when I start the Docker daemon:

ERROR: ../libkmod/libkmod.c:557 kmod_search_moddep() could not open moddep file '/lib/modules/4.4.0-0.bpo.tmw1-amd64/modules.dep.bin'

Docker shows this error 3 different times, when it runs modprobe bridge br_netfilter, nf_nat, and xt_conntrack.

**Comment by tonistiigi** [comment]
cc @boucher

**Comment by tswift242** [comment]
I'm also able to reproduce this on an AWS ec2 instance running a slightly different configuration:

```
Containers: 1
 Running: 1
 Paused: 0
 Stopped: 0
Images: 1
Server Version: 1.13.0-dev
Storage Driver: overlay
 Backing Filesystem: extfs
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: apparmor seccomp
Kernel Version: 4.4.19-040419-generic
Operating System: Debian GNU/Linux 8 (jessie) (containerized)
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 14.94 GiB
Name: e3fadcfe5803
ID: IEEA:P3UB:WIMT:765R:HJK5:DMS3:GUCY:FZNJ:2JVX:BGG3:D4ZX:RA2P
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 17
 Goroutines: 28
 System Time: 2016-10-06T20:42:26.829785814Z
 EventsListeners: 0
Registry: https://index.docker.io/v1/
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Comment by boucher** [comment]
Just to clarify: in all of these instances, are you talking about checkpointing from inside a docker container?

**Comment by tswift242** [comment]
Yes, checkpointing from inside a docker container; the standard docker-dev container to be precise.

**Comment by tswift242** [comment]
To me, it actually doesn't seem to be a C/R specific issue, as like I said above, just starting the Docker daemon leads to the very same error message being displayed that criu outputs. It seems more like an issue with the docker-dev environment.

**Comment by hodiapa** [comment]
Even in my case the checkpoint restore doesn't succeed. I am not running checkpoint inside the docker container though

**Comment by boucher** [comment]
Can you provide more details?

**Comment by hodiapa** [comment]
` docker checkpoint create --leave-running nas1 chk1
Error response from daemon: Cannot checkpoint container nas1: rpc error: code = 2 desc = exit status 1: "criu failed: type NOTIFY errno 0\nlog file: /var/lib/docker/containers/adfb66d5230eec64adf8dafaf80545431418113c7cbfcc2ff73a6f58ee0a5837/checkpoints/chk1/criu.work/dump.log\n"
`

dump file: https://gist.github.com/hodiapa/a74a5990fa08675cb81ab2ac1319a665

**Comment by boucher** [comment]
Checkpoint isn't currently supported for interactive containers, only ones
started in background mode.

**Comment by hodiapa** [comment]
Ok .. I did following:

```
vagrant@vagrant:~$ docker run -d --security-opt seccomp:unconfined --name nas1 spirals/npb3.3.1 -np 16 ep.C.16 
4b32577ba2ed2a583ced5482ef50cc63b3cb0b4067dad46385e5eb2de2ea5b7f
```

Error:

```
vagrant@vagrant:~$ docker checkpoint create --leave-running nas1 chk2
Error response from daemon: Cannot checkpoint container nas1: rpc error: code = 2 desc = exit status 1: "criu failed: type NOTIFY errno 0\nlog file: /var/lib/docker/containers/4b32577ba2ed2a583ced5482ef50cc63b3cb0b4067dad46385e5eb2de2ea5b7f/checkpoints/chk2/criu.work/dump.log\n"

```


dump: https://gist.github.com/hodiapa/2c46ba21f35153fbb2bc2f6804f7363b

**Comment by hodiapa** [comment]
@boucher any clues?

**Comment by boucher** [comment]
@hodiapa I would try emailing the CRIU mailing list, or filing an issue on the CRIU github repo -- I don't think this failure is related to the Docker integration. Here's the relevant line from your dump, in case it gives you any ideas:

```
(00.480204) Error (files-reg.c:683): Can't dump ghost file /dev/shm/mpich_shar_tmpGj4i0H of 82859216 size, increase limit
(00.480209) Error (cr-dump.c:1365): Dump mappings (pid: 28053) failed with -1
```
